src/Schedule.cpp

```diff
@@ -12,17 +12,21 @@ namespace Internal {
 struct LoopLevelContents {
     mutable RefCount ref_count;
 
-    // Note: func_ is empty for inline or root.
+    // Note: func_name is "" for inline or root.
     std::string func_name;
+    // If set to -1, this loop level does not refer to a particular stage of the
+    // function. 0 refers to initial stage, 1 refers to the 1st update stage, etc.
+    int stage;
```

Comment:
Would prefer "stage_index" I think. I'm not 100% sure about this however. The issue is that "Stage" is a type and it feels like "stage()" should return a Stage. I think part of my unease here is I'm worried about using the index into the stage list at all, as it feels fragile, but I don't really have a better idea.